syntax = "proto3";

package fashion;

service CustomerService {
  rpc CreateCustomer(Customer) returns (Customer) {}
  rpc ReadCustomer(CustomerId) returns (Customer) {}
  rpc UpdateCustomer(Customer) returns (Customer) {}
  rpc DeleteCustomer(CustomerId) returns (Customer) {}
  rpc ListCustomers(Empty) returns (stream Customer) {}
}

message Customer {
  int32 id = 1;
  string name = 2;
  string email = 3;
  string phone = 4;
}

message CustomerId {
  int32 id = 1;
}

message Empty {}
